api/src/main/java/org/openmrs/hl7/impl/HL7ServiceImpl.java

```diff
@@ -539,12 +539,10 @@ public Integer resolveLocationId(PL pl) throws HL7Exception {
 		// HACK: try to treat the first component (which should be "Point of
 		// Care" as an internal openmrs location_id
 		try {
-			Integer locationId = Integer.valueOf(pointOfCare);
-			Location l = Context.getLocationService().getLocation(locationId);
-			if (l != null) {
-				return l.getLocationId();
+			 Integer locationId = Integer.valueOf(pointOfCare);
+	            if (locationId == null) throw new Exception();
```

Comment:
@jwnasambu Its better to add a proper message to the exception mentioning the reason

Reply:
@Herbert did you have a look at my comment above?

Reply:
@dilanthas am not getting you

Reply:
@jwnasambu I was suggesting to add a message to the exception. something like **throw new Exception("Invalid location id");**